src/polysolve/LinearSolverSympiler.hpp

```diff
@@ -0,0 +1,59 @@
+#ifdef POLYSOLVE_WITH_SYMPILER
+
+#include <polysolve/LinearSolver.hpp>
+#include <sympiler/parsy/cholesky_solver.h>
+#include <Eigen/Sparse>
+#include <Eigen/Core>
+
+namespace polysolve
+{
+    class LinearSolverSympiler : public LinearSolver
+    {
+    public:
+        LinearSolverSympiler() = default;
+        ~LinearSolverSympiler() = default;
+
+    private:
+        POLYSOLVE_DELETE_MOVE_COPY(LinearSolverSympiler)
+
+    protected:
+        void setSolverMode(int _solver_mode);
+
+    public:
+        //////////////////////
+        // Public interface //
+        //////////////////////
+
+        // Set solver parameters
+        virtual void setParameters(const json &params) override;
+
+        // Retrieve status (success/failure) of factorization
+        virtual void getInfo(json &params) const override;
+
+        // Analyze sparsity pattern
+        virtual void analyzePattern(const StiffnessMatrix &A, const int precond_num) override;
+
+        // Factorize system matrix
+        virtual void factorize(const StiffnessMatrix &A) override;
+
+        // Solve the linear system Ax = b
+        virtual void solve(const Ref<const VectorXd> b, Ref<VectorXd> x) override;
+
+        // Name of the solver type (for debugging purposes)
+        virtual std::string name() const override { return "Sympiler"; }
+
+    private:
+        sym_lib::parsy::SolverSettings *solver_ = nullptr;
+        sym_lib::parsy::CSC *A_csc_ = new sym_lib::parsy::CSC;
```

Comment:
I think both of these objects cause memory leaked as they are never deleted. Consider using a smart pointer instead. Based on the semantics, it seem [value_ptr](https://github.com/adobe/lagrange/blob/main/modules/core/include/lagrange/utils/value_ptr.h) is a good candidate.

Reply:
Refactored to use smart pointers